portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,22 +24,35 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/literature")
+/** Servlet that maintains the comment section.*/
+@WebServlet("/comment")
 public final class DataServlet extends HttpServlet {
 
-  private List<String> BookRecs;
+  private CommentHistory comments = new CommentHistory();
+
   @Override
-  public void init() {
-    BookRecs = new ArrayList<>();
-    BookRecs.add("The Thread by Victoria Hislop");
-    BookRecs.add("The Hunchback of Notre Dame by Victor Hugo");
-    BookRecs.add("The Brothers Karamazov by Fyodor Dostoevsky");
+  /** Load all comment history onload **/
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String json = new Gson().toJson(comments);
+    System.out.println("got json string " + json);
+    response.getWriter().println(json);   
   }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String book = BookRecs.get((int) (Math.random() * BookRecs.size()));
-    response.setContentType("text/html;");
-    response.getWriter().println(book);
+  /** Add comment submission to the comments object **/
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-input", "");
+    comments.addComment(text);
+    response.sendRedirect("/index.html");
+  }
+
+  /** Safe wrapper for getParameter **/
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
```

Comment:
i know this was from the sample code they gave us but from what i've been told having more than a single exit/return statement from a method isn't generally considered good design.

Reply:
Duly noted, thanks!

Reply:
This is sometimes advised for very long or convoluted functions but in general, multiple return statements make the code more readable as they tend to reduce unnecessary long if-then-else chains and make it easier to reason about and track what is happening in the code. 

Also, removing return statements in fact makes the problem worse sometimes as now you will mostly instead end up using a local variable to store the final retval, except that variable can be modified anywhere in the function, so you have to keep track of it all throughout to figure out what would happen to it.